From 7bff1934de70913b6ab965510baa07059b7e8b25 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Sun, 10 Aug 2014 18:20:35 -0400
Subject: [iommu] Change iommu driver to call attach_device_to_domain trace event

Message-id: <20140810182035.24055.75709.stgit@gir.stowe>
Patchwork-id: 87582
O-Subject: [RHEL7.1 PATCH 11/19] iommu: Change iommu driver to call attach_device_to_domain trace event
Bugzilla: 1127467
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Dutile <ddutile@redhat.com>
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1127467
Upstream Status: b54db778858bc83f9231e5b358cb978f559f7016

commit b54db778858bc83f9231e5b358cb978f559f7016
Author: Shuah Khan <shuah.kh@samsung.com>
Date:   Thu Aug 15 11:59:26 2013 -0600

    iommu: Change iommu driver to call attach_device_to_domain trace event

    Change iommu driver to call attach_device_to_domain trace event. This
    iommu_device class event can be enabled to trigger when devices are attached
    to a domain. Trace information includes device name.

    Testing:
    Added trace calls to iommu_prepare_identity_map() for testing some of the
    conditions that are hard to trigger. Here is the trace from the testing:

          swapper/0-1     [003] ....     1.854102: attach_device_to_domain: IOMM

    Signed-off-by: Shuah Khan <shuah.kh@samsung.com>
    Signed-off-by: Joerg Roedel <joro@8bytes.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/iommu/iommu.c b/drivers/iommu/iommu.c
index 278055b..74c371c 100644
--- a/drivers/iommu/iommu.c
+++ b/drivers/iommu/iommu.c
@@ -685,10 +685,14 @@ EXPORT_SYMBOL_GPL(iommu_domain_free);
 
 int iommu_attach_device(struct iommu_domain *domain, struct device *dev)
 {
+ int ret;
  if (unlikely(domain->ops->attach_dev == NULL))
   return -ENODEV;
 
- return domain->ops->attach_dev(domain, dev);
+ ret = domain->ops->attach_dev(domain, dev);
+ if (!ret)
+  trace_attach_device_to_domain(dev);
+ return ret;
 }
 EXPORT_SYMBOL_GPL(iommu_attach_device);
 
-- 
1.7.1